firmware/pico/src/app/hal.hh

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <cstdint>
+
+uint64_t get_time_since_boot_us();
+uint32_t get_time_since_boot_ms();
+
+class ClockSource {
+   public:
+    static ClockSource& instance();
+    ~ClockSource();
+
+    uint64_t get48MHzTickCount();
```

Comment:
Recommend renaming to Get48MHzTickCount(). The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Function_Names) allows getters and setters to be named like variables, but for now I think it is nicest to keep things consistent across the codebase.